test/transaction-optimistic.test.ts

```diff
@@ -0,0 +1,24 @@
+import { describe, expect, it } from 'vitest';
+import { RocksDatabase, Transaction } from '../src/index.js';
+import { dbRunner } from './lib/util.js';
+describe('Optimistic Transaction Specific Behavior', () => {
+	it(`async should be able to resume a conflicted transaction`, () => dbRunner({
+	}, async ({ db }) => {
+		await db.put('foo', 'bar1');
+
+		setTimeout(() => db?.put('foo', 'bar2'));
+
+		let transaction = new Transaction(db.store);
```

Comment:
nit: consider `const`?

```suggestion
		const transaction = new Transaction(db.store);
```